DFC.TestAutomation.UI/Helper/CommonActionHelper.cs

```diff
@@ -0,0 +1,239 @@
+        /// is trimmed before being compared.
+        /// </summary>
+        /// <param name="locator">The IWebElement locator.</param>
+        /// <param name="expectedText">The expected text.</param>
+        /// <returns>True if the IWebElement contains the expected text. False if the IWebElement does not contain the expected text.</returns>
+        public bool ElementContainsText(By locator, string expectedText)
+        {
+            bool Action()
+            {
+                this.WebDriverWaitHelper.WaitForPageToLoad();
+                var actual = this.GetText(locator);
+                return actual.Contains(expectedText.Trim(), StringComparison.CurrentCultureIgnoreCase);
+            }
+
+            return this.RetryHelper.RetryOnException(Action);
+        }
+
+        /// <summary>
+        /// Gets the attribute value for an IWebElement.
+        /// </summary>
+        /// <param name="locator">The IWebElement locator.</param>
+        /// <param name="attributeName">The attribute name.</param>
+        /// <returns>The atrribute value.</returns>
+        public string GetAttributeValue(By locator, string attributeName)
+        {
+            string Action()
+            {
+                return this.WebDriver.FindElement(locator).GetAttribute(attributeName);
+            }
+
+            return this.RetryHelper.RetryOnException<string>(Action);
+        }
+
+        /// <summary>
+        /// Gets the text value from all IWebElements with a given locator.
+        /// </summary>
+        /// <param name="locator">The IWebELement locator.</param>
+        /// <returns>The string value.</returns>
+        public string GetTextFromElements(By locator)
+        {
+            var text = string.Empty;
+            IList<IWebElement> webElementGroup = this.WebDriver.FindElements(locator);
+
+            foreach (IWebElement webElement in webElementGroup)
+            {
+                text += this.GetText(webElement);
+            }
+
+            return text;
+        }
+
+        /// <summary>
+        /// Gets the count of the IWebElement collection returned when locating all IWebElements with the given locator.
+        /// </summary>
+        /// <param name="locator">The IWebElement locator.</param>
+        /// <returns>An integer indicating how many IWebElements are located using the locator.</returns>
+        public int GetCountOfElements(By locator)
+        {
+            return this.WebDriver.FindElements(locator).Count;
+        }
+
+        /// <summary>
+        /// Assesses whether an IWebElement is present.
+        /// </summary>
+        /// <param name="locator">The IWebElement locator.</param>
+        /// <returns>A value indicating whether the IwebElement is present.</returns>
+        public bool IsElementPresent(By locator)
+        {
+            this.WebDriverWaitHelper.SetImplicitWait(500);
```

Comment:
@jbarnes16503 Why are we specifically setting the implicitWait to 500ms for these two functions? (Aren't we using a general configuration to set this level within the WebDriverWaitHelper.cs file? `this.WebDriverImplicitWait = new WebDriverWait(webDriver, TimeSpan.FromSeconds(this.TimeoutConfiguration.ImplicitWait));`

Reply:
We have 2 'is the element present?' methods:

1. WaitForElementToBePresent() in the WebDriverWaitHelper
2. IsElementPresent() a you have indicated above

There are cases where you need the element in order to proceed with your tests in which case you'd use number 1 above. If for example your user journey depends on the presence of an object (a cookie banner for example) then you'd want to simply interrogate the page to see if the object is there. If we didn't reset the timeout and the object was not there, then it would potentially keep looking for the implicit wait length of time. This essentially result in a 60 second delay which wouldn't add any additional value.

So In short think of number 1 being 'wait until its there' and number 2 being 'is it there right now?'. I hope that makes sense.

Reply:
I does make sense ...but my question was more about the decision to fix this wait at 500. _Again, happy to leave and refine as appropriate if a need present itself._

Reply:
I see. No we don't want to do that as the explicit wait (which is applied globally) is set in the users appsettings.json. So we only want to override this in a handful of places and only in cases where we are looking for immeadiate feedback.

Reply:
I could have set the value to 0 but 500 is half a second and from experience it's always sensible to leave a little wait time. But yes if we find this isn't working for us we have plenty of options.